### PR TITLE
Using conda metapackage when installing cuda 11.6 and 11.7

### PR DESCRIPTION
**Pull request by atalman**
Following PR: https://github.com/pytorch/builder/pull/1094
Adds cuda metapackage called pytorch-cuda . This way we can make sure to install correct version of cuda dependencies and don't depend on conda-forge.

### CODE REVIEW

File: .circleci/unittest/linux/scripts/install.sh [hunk 1]
```diff
@@ -23,10 +23,13 @@ else
     version="$(python -c "print('.'.join(\"${CUDA_VERSION}\".split('.')[:2]))")"
 
     cuda_toolkit_pckg="cudatoolkit"
+    cudatoolkit=""
     if [[ "$CU_VERSION" == cu116 ]]; then
         cuda_toolkit_pckg="cuda"
+        cudatoolkit="-c nvidia cuda=11.6 atalman::pytorch-cuda=1.0=cuda11.6"
```
Comment (datumbox) [inline .circleci/unittest/linux/scripts/install.sh:29]: Your personal username appears here. Is that accidental or perhaps at temporary workaround?

File: packaging/build_cmake.sh [hunk 2]
```diff
@@ -42,7 +42,12 @@ else
   PYTORCH_MUTEX_CONSTRAINT=''
 fi
 
-conda install -yq \pytorch=$PYTORCH_VERSION $CONDA_CUDATOOLKIT_CONSTRAINT $PYTORCH_MUTEX_CONSTRAINT $MKL_CONSTRAINT numpy -c nvidia -c "pytorch-${UPLOAD_CHANNEL}"
+if [[ $CU_VERSION == "cu116" ]]; then
+    conda install -yq \pytorch=$PYTORCH_VERSION  -c nvidia cuda=11.6 atalman::pytorch-cuda=1.0=cuda11.6 $PYTORCH_MUTEX_CONSTRAINT $MKL_CONSTRAINT numpy -c nvidia -c "pytorch-${UPLOAD_CHANNEL}"
```
Comment (datumbox) [inline packaging/build_cmake.sh:46]: Same here.

**Review by datumbox — COMMENTED** [review]
@atalman Thanks for the PR, I understand that this is still in progress. Just a quick question:

File: .circleci/unittest/linux/scripts/install.sh (hunk 1 above)
Comment (atalman) [reply .circleci/unittest/linux/scripts/install.sh:29]: @datumbox This is testing of this package: https://anaconda.org/atalman/pytorch-cuda/files . Here is the PR that introduces this metapackage https://github.com/pytorch/builder/pull/1094

Comment (datumbox) [reply .circleci/unittest/linux/scripts/install.sh:29]: Gotcha, I understand it will be moved to pytorch once you are done. Thanks for clarifying.

Comment (malfet) [inline .circleci/unittest/linux/scripts/install.sh:29]: Why do you need `cuda=11.6` package here? 
```suggestion
        cudatoolkit="-c nvidia atalman::pytorch-cuda=1.0=cuda11.6"
```

Comment (atalman) [reply .circleci/unittest/linux/scripts/install.sh:29]: Basically metapackage just sets constraints. Selecting only metapackage does not work

```
conda install -c nvidia -c pytorch-nightly pytorch-cuda=1.0=cuda11.6
Collecting package metadata (current_repodata.json): done
Solving environment: done

## Package Plan ##

  environment location: /home/atalman/anaconda3/envs/mpy3.91

  added / updated specs:
    - pytorch-cuda==1.0=cuda11.6


The following packages will be downloaded:

    package                    |            build
    ---------------------------|-----------------
    pytorch-cuda-1.0           |         cuda11.6           6 KB  pytorch-nightly
    ------------------------------------------------------------
                                           Total:           6 KB

The following NEW packages will be INSTALLED:

  pytorch-cuda       pytorch-nightly/noarch::pytorch-cuda-1.0-cuda11.6


Proceed ([y]/n)? n
```

**Review by datumbox — APPROVED** [review]
LGTM, thanks a lot for the work @atalman.

**Pull request merged**